libr/util/utype.c

```diff
@@ -766,3 +767,230 @@ R_API char *r_type_func_name(Sdb *types, const char *fname) {
 	}
 	return r_type_func_guess (types, fname);
 }
+
+static bool parse_bits_suffix(const char *name, const char *prefix, int *size_out) {
+	size_t plen = strlen (prefix);
+	if (r_str_ncasecmp (name, prefix, plen) != 0) {
+		return false;
+	}
+	const char *rest = name + plen;
+	size_t rlen = strlen (rest);
+	if (rlen >= 2 && !r_str_casecmp (rest + rlen - 2, "_t")) {
+		rlen -= 2;
+	}
+	if (rlen == 0) {
+		return false;
+	}
+	size_t i;
+	for (i = 0; i < rlen; i++) {
+		if (!isdigit ((unsigned char)rest[i])) {
+			return false;
+		}
+	}
+	char *tmp = r_str_ndup (rest, rlen);
+	int bits = atoi (tmp);
+	free (tmp);
+	if (bits <= 0 || (bits % 8) != 0) {
+		return false;
+	}
+	*size_out = bits / 8;
+	return true;
+}
+
+R_API RTypeCTypeInfo *r_type_parse_ctype(const char *ctype, int ptr_size, int long_size, int int_size) {
+	R_RETURN_VAL_IF_FAIL (ctype, NULL);
+	char *lower = r_str_trim_dup (ctype);
+	if (R_STR_ISEMPTY (lower)) {
+		free (lower);
+		return NULL;
+	}
+	r_str_case (lower, false);
+
+	RTypeCTypeInfo *info = R_NEW0 (RTypeCTypeInfo);
+	if (!info) {
+		free (lower);
+		return NULL;
+	}
+
+	if (!strcmp (lower, "size_t") || !strcmp (lower, "uintptr_t")) {
```

Comment:
maybe all those checks can be defined in an array of structs with string+attributes, and just loop over it